### PR TITLE
Add ability to fix RNG state so that folds from IterativeStratification are reproducible

### LINKED ISSUE #144: Random_state does not ensure reproducible splits in stratified kfold
Stratified iterative kfold does not return the same when we run it twice  using random_state = 0.
Here is a working exemple:

```
from skmultilearn.model_selection import IterativeStratification
import numpy as np
import copy
np.random.seed(0)
Y0 = np.random.choice([0,1], 100)[:, np.newaxis]
Y1 = np.random.choice([0,1], 100)[:, np.newaxis]
Y2 = np.random.choice([0,1], 100)[:, np.newaxis]

Y = np.hstack((Y0, Y1, Y2))
X = None 

random_state = 0
skf = IterativeStratification(10, random_state=random_state)
indTrain = ()
indTest = ()
X = np.zeros(Y.shape)
for igen in skf.split(X, Y):
    indTrain += (igen[0],)
    indTest += (igen[1],)

tr_save = copy.deepcopy(indTrain)
te_save = copy.deepcopy(indTrain)

indTrain = ()
indTest = ()
for i, igen in enumerate(skf.split(X, Y)):
    np.testing.assert_array_equal(igen[0], tr_save[i])
    np.testing.assert_array_equal(igen[1], te_save[i])
```

I suspect the problem comes from function def _fold_tie_break in iterative_stratification.py line 124.
random choice is used here but no call to random_state is done.

Another caveat appear in line 311  in function _distribute_negative_evidence:
fold_selected = np.random.choice(np.where(self.desired_samples_per_fold > 0)[0], 1)[0]

Random choice is called without calling random state first with np.random.seed(random_state).

I am using python 3.6 on windows with pycharm. I installed skmultilearn with pip (v0.2.0).

What do you think?

### PR DESCRIPTION
**Pull request by x0wllaar**
Currently, it is impossible to pass an RNG seed to IterativeStratification, which makes getting reproducible results from it impossible. This PR exposes the shuffle parameter of the base class in the IterativeStratification constructor. It also makes some changes that allow the CV results to become reproducible.  

Notably, it changes all the np.random.choice calls from within IterativeStratification to use the RNG seeded in the constructor (or the global NumPy RNG if the seed is none). It also makes some changes to the _fold_tie_break function to allow it to use the RNG state.

These changes make the folds produced by IterativeStratification reproducible if one passes random_state to the constructor.

This should fix #144. I also should mention that credit for investigating the causes of non-reproducibility should go to @VaelK and @blackcat84 (see #144 )

### CODE REVIEW

**Comment by nsarang** [comment]
Is this package not maintained anymore?

**Comment by dyhan316** [comment]
Hello @x0wllaar ! I tried to do `IterativeStratification(n_splits= SPLITS, order=10, random_state = np.random.seed(0))` but the nI call it multiple times, each time the result is different. Could you please provide an example of how to properly use your version of the code?

**Comment by ChristianSch** [comment]
Woops, this broke the PR somehow 🙈

File: skmultilearn/model_selection/iterative_stratification.py [hunk 1]
```diff
@@ -347,9 +357,6 @@ def _iter_test_indices(self, X, y=None, groups=None):
         fold : List[int]
             indexes of test samples for a given fold, yielded for each of the folds
         """
-        if self.random_state:
-            check_random_state(self.random_state)
```
Comment (ChristianSch) [inline skmultilearn/model_selection/iterative_stratification.py:351]: is done in constructor now

File: skmultilearn/model_selection/iterative_stratification.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def iterative_train_test_split(X, y, test_size, random_state=None):
 
 
 
-def _fold_tie_break(desired_samples_per_fold, M, random_state=None):
+def _fold_tie_break(desired_samples_per_fold, M, rng_state=check_random_state(None)):
```
Comment (ChristianSch) [inline skmultilearn/model_selection/iterative_stratification.py:106]: not sure we should deviate from the naming. any reason for that?

File: skmultilearn/model_selection/iterative_stratification.py [hunk 3]
```diff
@@ -321,7 +327,11 @@ def _distribute_negative_evidence(self, rows_used, folds):
             row = available_samples.pop()
             rows_used[row] = True
             samples_left -= 1
-            fold_selected = np.random.choice(np.where(self.desired_samples_per_fold > 0)[0], 1)[0]
+            fold_selected = self._
+            
+            
+            
+            state.choice(np.where(self.desired_samples_per_fold > 0)[0], 1)[0]
```
Comment (ChristianSch) [inline skmultilearn/model_selection/iterative_stratification.py:334]: gotta fix this

**Review by ChristianSch — CHANGES_REQUESTED** [review]

File: skmultilearn/model_selection/iterative_stratification.py (hunk 2 above)
Comment (x0wllaar) [reply skmultilearn/model_selection/iterative_stratification.py:106]: No, I will fix that, thank you

File: skmultilearn/model_selection/iterative_stratification.py (hunk 3 above)
Comment (x0wllaar) [reply skmultilearn/model_selection/iterative_stratification.py:334]: Yes, thank you! Not sure how that leaked into the PR

**Comment by x0wllaar** [comment]
I pushed the required changes to my branch. I didn't quite understand what I need to change for the first part of the review about the constructor though :(.

File: skmultilearn/model_selection/iterative_stratification.py (hunk 2 above)
Comment (x0wllaar) [reply skmultilearn/model_selection/iterative_stratification.py]: I think that I had something in mind while I was working on it, I changed this to the correct version

**Review by ChristianSch — APPROVED** [review]

**Comment by ChristianSch** [comment]
No worries, that was just a note to myself haha. I don't have the ability to test it, but I'll just trust it works
